scripts/scenario_mat.lua

```diff
@@ -1055,7 +1056,24 @@ end
 
 function setScenario(params)
     CurrentScenario = params
-    updateAssistant("POST", "setScenario", params)
+    Async(function()
+        local res = coroutine.yield(function (resolve)
```

Comment:
I'm quite confused with what resolve is - where it comes from-, maybe add some comments ?